ext/msgpack/buffer_class.c

```diff
@@ -53,16 +46,40 @@ static void Buffer_free(void* data)
         return;
     }
     msgpack_buffer_t* b = (msgpack_buffer_t*) data;
+    if (RTEST(b->owner)) {
+        // This buffer is embded in a packer or unpacker, the actual owner
+        // will free it.
+        return;
+    }
     msgpack_buffer_destroy(b);
     xfree(b);
 }
 
+const rb_data_type_t buffer_data_type = {
+    .wrap_struct_name = "msgpack:buffer",
+    .function = {
+        .dmark = msgpack_buffer_mark,
+        .dfree = Buffer_free,
+        .dsize = NULL, // TODO: memsie func would be nice
```

Comment:
I need to look at how to measure this properly. I'd prefer to do it in a followup though, it's not a regression since the old API wouldn't provide it either.